oozie/src/main/java/org/apache/falcon/workflow/engine/OozieWorkflowEngine.java

```diff
@@ -1434,34 +1434,40 @@ private Frequency getDelay(Feed entity, CoordinatorJob coord) {
     }
 
     private String updateInternal(Entity oldEntity, Entity newEntity, Cluster cluster, BundleJob oldBundle,
-        String user, Boolean skipDryRun) throws FalconException {
+                                  String user, Boolean skipDryRun) throws FalconException {
+        String currentUser = CurrentUser.getUser();
+        switchUser(user);
+
         String clusterName = cluster.getName();
 
         Date effectiveTime = getEffectiveTime(cluster, newEntity);
         LOG.info("Effective time " + effectiveTime);
+        try {
+            //Validate that new entity can be scheduled
+            dryRunForUpdate(cluster, newEntity, effectiveTime, skipDryRun);
+
+            boolean suspended = BUNDLE_SUSPENDED_STATUS.contains(oldBundle.getStatus());
+
+            //Set end times for old coords
+            updateCoords(clusterName, oldBundle, EntityUtil.getParallel(oldEntity), effectiveTime, newEntity);
+            //schedule new entity
+            String newJobId = scheduleForUpdate(newEntity, cluster, effectiveTime, user);
+            BundleJob newBundle = null;
+            if (newJobId != null) {
+                newBundle = getBundleInfo(clusterName, newJobId);
+            }
 
-        //Validate that new entity can be scheduled
-        dryRunForUpdate(cluster, newEntity, effectiveTime, skipDryRun);
-
-        boolean suspended = BUNDLE_SUSPENDED_STATUS.contains(oldBundle.getStatus());
-
-        //Set end times for old coords
-        updateCoords(clusterName, oldBundle, EntityUtil.getParallel(oldEntity), effectiveTime, newEntity);
-
-        //schedule new entity
-        String newJobId = scheduleForUpdate(newEntity, cluster, effectiveTime, user);
-        BundleJob newBundle = null;
-        if (newJobId != null) {
-            newBundle = getBundleInfo(clusterName, newJobId);
-        }
-
-        //Sometimes updateCoords() resumes the suspended coords. So, if already suspended, resume now
-        //Also suspend new bundle
-        if (suspended) {
-            doBundleAction(newEntity, BundleAction.SUSPEND, cluster.getName());
+            //Sometimes updateCoords() resumes the suspended coords. So, if already suspended, resume now
+            //Also suspend new bundle
+            if (suspended) {
+                doBundleAction(newEntity, BundleAction.SUSPEND, cluster.getName());
+            }
+            switchUser(currentUser);
```

Comment:
You don't need to call "switchUser" here. You already add it to the "finally" block :)